app/src/main/java/org/schabi/newpipe/local/subscription/SubscriptionFragment.kt

```diff
@@ -150,28 +135,71 @@ class SubscriptionFragment : BaseStateFragment<SubscriptionState>() {
 
         activity.supportActionBar?.setDisplayShowTitleEnabled(true)
         activity.supportActionBar?.setTitle(R.string.tab_subscriptions)
+
+        buildImportExportMenu(menu)
     }
 
-    private fun setupBroadcastReceiver() {
-        if (activity == null) return
+    private fun buildImportExportMenu(menu: Menu) {
+        // -- Import --
+        val importSubMenu = menu.addSubMenu(R.string.import_from)
 
-        if (subscriptionBroadcastReceiver != null) {
-            LocalBroadcastManager.getInstance(activity).unregisterReceiver(subscriptionBroadcastReceiver!!)
-        }
+        addMenuItemToSubmenu(importSubMenu, R.string.previous_export) { onImportPreviousSelected() }
+            .setIcon(R.drawable.ic_backup)
+
+        val services = requireContext().resources.getStringArray(R.array.service_list)
+        for (serviceName in services) {
+            try {
+                val service = NewPipe.getService(serviceName)
+
```

Comment:
You can use `for (service in ServiceList.all()) {` and avoid handling strings and avoid the `try`. The `service_list` seems to be completely outdated, please remove it altogether (it is not used in any other place):

![image](https://user-images.githubusercontent.com/36421898/167290335-54b60fde-bfb1-47ff-be8a-a67d4459ed64.png)

<sup>Sorry for not noticing this earlier</sup>

Reply:
doen